lib/client.js

```diff
@@ -263,7 +263,23 @@ Client.prototype._buildForward = function() {
       }
 
       if (qos && options.messageId) {
-        that.server.updateOfflinePacket(that, options.messageId, packet, doForward);
+
+        //converting the payload to a JSON object
+        var packetPayload = JSON.parse(packet.payload.toString());
```

Comment:
I'm against forcing clients to send the messageId inside the payload, and also forcing a payload format in Mosca. MQTT does not mandate any format.